### PR TITLE
Fluent: Add motion variables, mixins, and classes

### PR DESCRIPTION
**Pull request by mikewheaton**
Adds variables, mixins, and classes for the new Fluent motions. All of the existing animation variables, mixins, and classes have been moved to files marked `MDL2` for deprecation. These are imported by the new files to preserve compatibility.

Note that the new items are named 'motion' rather than 'animation'. This matches the language used by the designers and allows MDL2 and Fluent to coexist without naming conflicts.

### CODE REVIEW

File: src/sass/mixins/_Motion.Mixins.scss [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+// Slides.
+@mixin ms-motion-slideLeftOut {
+  @include ms-LTR {
+    animation: ms-motion-slideLeftOut 100ms $ms-motion-timing-accelerate both,
+      ms-motion-fadeOut 100ms $ms-motion-timing-linear both;
+  }
+
+  @include ms-RTL {
```
Comment (mikewheaton) [inline src/sass/mixins/_Motion.Mixins.scss]: For horizontal slides, the direction is reversed for right-to-left (RTL).

**Comment by Jahnp** [comment]
Just a general thought: There are a few renames here (Animation --> Motion, Animation --> Animation.MDL2) that will break downstream consumers if they're looking for those files. Even though we don't document them explicitly, we know they're often used this way. We should be careful about renames on branches we know are in active use.

We should consider re-exporting the new variable names from files with the old names (e.g. retaining the `_Animation` files) until we've done a major release containing the change.

**Comment by mikewheaton** [comment]
I've renamed the files from Motion to Animation to maintain compatibility for projects that are importing partials rather than using References.scss.

File: src/sass/variables/_Animation.Variables.MDL2.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE in the project root for license information.
+
```
Comment (Jahnp) [inline src/sass/variables/_Animation.Variables.MDL2.scss]: I know it's implied by the filename, but we should add a comment in the file that these are now deprecated

Comment (mikewheaton) [reply src/sass/variables/_Animation.Variables.MDL2.scss]: Added

**Pull request merged**